common/src/main/java/com/microsoft/identity/common/internal/eststelemetry/EstsTelemetry.java

```diff
@@ -194,122 +215,143 @@ private IRequestTelemetryCache createLastRequestTelemetryCache(@NonNull final Co
         return new SharedPreferencesLastRequestTelemetryCache(sharedPreferencesFileManager);
     }
 
-    private RequestTelemetry setupLastFromCurrent(@Nullable RequestTelemetry currentTelemetry) {
-        if (currentTelemetry == null) {
-            return new RequestTelemetry(Schema.CURRENT_SCHEMA_VERSION, false);
-        }
-
-        RequestTelemetry lastTelemetry = new RequestTelemetry(currentTelemetry.getSchemaVersion(), false);
-
-        // grab whatever common fields we can from current request
-        for (Map.Entry<String, String> entry : currentTelemetry.getCommonTelemetry().entrySet()) {
-            lastTelemetry.putTelemetry(entry.getKey(), entry.getValue());
-        }
-
-        // grab whatever platform fields we can from current request
-        for (Map.Entry<String, String> entry : currentTelemetry.getPlatformTelemetry().entrySet()) {
-            lastTelemetry.putTelemetry(entry.getKey(), entry.getValue());
-        }
-
-        return lastTelemetry;
-    }
-
     /**
      * Flush the telemetry data for the current request to the {@link android.content.SharedPreferences} using the {@link SharedPreferencesLastRequestTelemetryCache}.
      * Removes the telemetry associated to the correlation id from the telemetry map,
      * and saves it to the cache (SharedPreferences) as the last request telemetry.
      */
-    void flush() {
-        String correlationId = DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID);
-        flush(correlationId);
-    }
-
-    /**
-     * Flush the telemetry data associated to the correlation id to the {@link android.content.SharedPreferences} using the {@link SharedPreferencesLastRequestTelemetryCache}.
-     * Removes the telemetry associated to the correlation id from the telemetry map,
-     * and saves it to the cache (SharedPreferences) as the last request telemetry.
-     *
-     * @param correlationId correlation id of the request
-     */
-    public void flush(final String correlationId) {
-        final String errorCode = null; // there was no error
-        flush(correlationId, errorCode);
-    }
-
-    /**
-     * Flush the telemetry data associated to the correlation id to the {@link android.content.SharedPreferences} using the {@link SharedPreferencesLastRequestTelemetryCache}.
-     * Removes the telemetry associated to the correlation id from the telemetry map,
-     * and saves it to the cache (SharedPreferences) as the last request telemetry.
-     *
-     * @param correlationId correlation id of the request
-     * @param baseException exception that may have occurred during the request
-     */
-    public void flush(final String correlationId, final BaseException baseException) {
-        flush(correlationId, baseException == null ? null : baseException.getErrorCode());
-    }
+    public void flush(@NonNull final BaseCommand command, @NonNull final CommandResult commandResult) {
```

Comment:
is this thread safe? (Since we support parallel request)

Reply:
Yes, we are using concurrent data structures.

Reply:
What I meant was... is it technically possible for any two concurrent requests to send the same set of 'failed_requests' or 'errors'. If so, would the server side be able to deal with it.

Reply:
Yes, it is possible, and the server won't do anything. It means that we would have duplicated data there.